src/vivarium_gates_iv_iron/components/observers.py

```diff
@@ -1,36 +1,255 @@
+    get_deaths,
     get_group_counts,
     get_output_template,
-    get_deaths,
-    get_state_person_time,
-    get_transition_count,
+    get_person_time,
     get_years_lived_with_disability,
     get_years_of_life_lost,
-    get_person_time,
-    QueryString,
-    TransitionString,
 )
 
-from vivarium_gates_iv_iron.constants import models, results, data_keys, data_values
+from vivarium_gates_iv_iron.constants import data_keys, data_values, models, results
+
+
+class LegacyMortalityObserver:
+    """(From vivarium_public_health 0.10.15) An observer for cause-specific deaths, ylls, and total person time.
+
+    By default, this counts cause-specific deaths, years of life lost, and
+    total person time over the full course of the simulation. It can be
+    configured to bin these measures into age groups, sexes, and years
+    by setting the ``by_age``, ``by_sex``, and ``by_year`` flags, respectively.
+
+    In the model specification, your configuration for this component should
+    be specified as, e.g.:
+
+    .. code-block:: yaml
+
+        configuration:
+            metrics:
+                mortality:
+                    by_age: True
+                    by_year: False
+                    by_sex: True
+
+    """
+
+    configuration_defaults = {
+        "metrics": {
+            "mortality": {
+                "by_age": False,
+                "by_year": False,
+                "by_sex": False,
+            }
+        }
+    }
+
+    @property
+    def name(self):
+        return "mortality_observer"
+
+    def setup(self, builder):
+        self.config = builder.configuration.metrics.mortality
+        self.clock = builder.time.clock()
+        self.step_size = builder.time.step_size()
+        self.start_time = self.clock()
+        self.initial_pop_entrance_time = self.start_time - self.step_size()
+        self.age_bins = get_age_bins(builder)
+        diseases = builder.components.get_components_by_type(
+            (DiseaseState, RiskAttributableDisease)
+        )
+        self.causes = [c.state_id for c in diseases] + ["other_causes"]
+
+        life_expectancy_data = builder.data.load(
+            "population.theoretical_minimum_risk_life_expectancy"
+        )
+        self.life_expectancy = builder.lookup.build_table(
+            life_expectancy_data, key_columns=[], parameter_columns=["age"]
+        )
+
+        columns_required = [
+            "tracked",
+            "alive",
+            "entrance_time",
+            "exit_time",
+            "cause_of_death",
+            "years_of_life_lost",
+            "age",
+        ]
+        if self.config.by_sex:
+            columns_required += ["sex"]
+        self.population_view = builder.population.get_view(columns_required)
+
+        builder.value.register_value_modifier("metrics", self.metrics)
+
+    def metrics(self, index, metrics):
+        pop = self.population_view.get(index)
+        pop.loc[pop.exit_time.isnull(), "exit_time"] = self.clock()
```

Comment:
I was having issues with time and when that happens on each time step.  Is using builder for clock better than event.time?

Reply:
There shouldn't be a difference, I don't think.